core/pkg/service/sync/sync.go

```diff
@@ -0,0 +1,139 @@
+package sync
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	rpc "buf.build/gen/go/open-feature/flagd/bufbuild/connect-go/sync/v1/syncv1connect"
+	syncv1 "buf.build/gen/go/open-feature/flagd/protocolbuffers/go/sync/v1"
+	"github.com/bufbuild/connect-go"
+	"github.com/open-feature/flagd/core/pkg/logger"
+	iservice "github.com/open-feature/flagd/core/pkg/service"
+	"github.com/open-feature/flagd/core/pkg/sync"
+	syncStore "github.com/open-feature/flagd/core/pkg/sync-store"
+	"golang.org/x/net/http2"
+	"golang.org/x/net/http2/h2c"
+)
+
+type SyncServer struct {
+	SyncStore     syncStore.SyncStore
+	server        http.Server
+	Configuration SyncServerConfiguration
+	Logger        *logger.Logger
+}
+
+type SyncServerConfiguration struct {
+	Port        uint16
+	MetricsPort uint16
+}
+
+func (s *SyncServer) Serve(ctx context.Context, svcConf iservice.Configuration) error {
+	mux := http.NewServeMux()
+	path, handler := rpc.NewFlagSyncServiceHandler(s)
```

Comment:
This is duplicated inside `setupServer()`

Reply:
removed